### ISSUE TITLE
Warnings...feel free to help me with suggestions

### ISSUE THREAD

**Issue by pusj** [pusj/LSTM-Time-Series-Analysis-using-Tensorflow#1]
Success consists of going from failure to failure without loss of enthusiasm.

Here are the warnings I get:

 RESTART: d\lstm_sin_rev1.py 
1
WARNING:tensorflow:From \AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\contrib\learn\python\learn\monitors.py:322: BaseMonitor.__init__ (from tensorflow.contrib.learn.python.learn.monitors) is deprecated and will be removed after 2016-12-05.
Instructions for updating:
Monitors are deprecated. Please use tf.train.SessionRunHook.
WARNING:tensorflow:From \AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\contrib\learn\python\learn\models.py:107: mean_squared_error_regressor (from tensorflow.contrib.learn.python.learn.ops.losses_ops) is deprecated and will be removed after 2016-12-01.
Instructions for updating:
Use `tf.contrib.losses.mean_squared_error` and explicit logits computation.
WARNING:tensorflow:From \AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\contrib\learn\python\learn\ops\losses_ops.py:39: mean_squared_error (from tensorflow.contrib.losses.python.losses.loss_ops) is deprecated and will be removed after 2016-12-30.
Instructions for updating:
Use tf.losses.mean_squared_error instead.
WARNING:tensorflow:From \AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\contrib\losses\python\losses\loss_ops.py:530: compute_weighted_loss (from tensorflow.contrib.losses.python.losses.loss_ops) is deprecated and will be removed after 2016-12-30.
Instructions for updating:
Use tf.losses.compute_weighted_loss instead.
WARNING:tensorflow:From \AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\contrib\losses\python\losses\loss_ops.py:151: add_loss (from tensorflow.contrib.losses.python.losses.loss_ops) is deprecated and will be removed after 2016-12-30.
Instructions for updating:
Use tf.losses.add_loss instead.

Warning (from warnings module):
  File "\AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\python\util\deprecation.py", line 247
    equality = a == b
FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
WARNING:tensorflow:From \AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\contrib\learn\python\learn\monitors.py:712: calling BaseEstimator.evaluate (from tensorflow.contrib.learn.python.learn.estimators.estimator) with y is deprecated and will be removed after 2016-12-01.
Instructions for updating:
Estimator is decoupled from Scikit Learn interface by moving into
separate class SKCompat. Arguments x, y and batch_size are only
available in the SKCompat class, Estimator will only accept input_fn.
Example conversion:
  est = Estimator(...) -> est = SKCompat(Estimator(...))
WARNING:tensorflow:From \AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\contrib\learn\python\learn\monitors.py:712: calling BaseEstimator.evaluate (from tensorflow.contrib.learn.python.learn.estimators.estimator) with x is deprecated and will be removed after 2016-12-01.
Instructions for updating:
Estimator is decoupled from Scikit Learn interface by moving into
separate class SKCompat. Arguments x, y and batch_size are only
available in the SKCompat class, Estimator will only accept input_fn.
Example conversion:
  est = Estimator(...) -> est = SKCompat(Estimator(...))
WARNING:tensorflow:From \AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\contrib\learn\python\learn\models.py:107: mean_squared_error_regressor (from tensorflow.contrib.learn.python.learn.ops.losses_ops) is deprecated and will be removed after 2016-12-01.
Instructions for updating:
Use `tf.contrib.losses.mean_squared_error` and explicit logits computation.
WARNING:tensorflow:From \AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\contrib\learn\python\learn\ops\losses_ops.py:39: mean_squared_error (from tensorflow.contrib.losses.python.losses.loss_ops) is deprecated and will be removed after 2016-12-30.
Instructions for updating:
Use tf.losses.mean_squared_error instead.
WARNING:tensorflow:From \AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\contrib\losses\python\losses\loss_ops.py:530: compute_weighted_loss (from tensorflow.contrib.losses.python.losses.loss_ops) is deprecated and will be removed after 2016-12-30.
Instructions for updating:
Use tf.losses.compute_weighted_loss instead.
WARNING:tensorflow:From \AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\contrib\losses\python\losses\loss_ops.py:151: add_loss (from tensorflow.contrib.losses.python.losses.loss_ops) is deprecated and will be removed after 2016-12-30.
Instructions for updating:
Use tf.losses.add_loss instead.
WARNING:tensorflow:Skipping summary for global_step, must be a float or np.float32.
(999, 1, 1)
(999, 1)
WARNING:tensorflow:From \AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\contrib\learn\python\learn\models.py:107: mean_squared_error_regressor (from tensorflow.contrib.learn.python.learn.ops.losses_ops) is deprecated and will be removed after 2016-12-01.
Instructions for updating:
Use `tf.contrib.losses.mean_squared_error` and explicit logits computation.
WARNING:tensorflow:From \AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\contrib\learn\python\learn\ops\losses_ops.py:39: mean_squared_error (from tensorflow.contrib.losses.python.losses.loss_ops) is deprecated and will be removed after 2016-12-30.
Instructions for updating:
Use tf.losses.mean_squared_error instead.
WARNING:tensorflow:From \AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\contrib\losses\python\losses\loss_ops.py:530: compute_weighted_loss (from tensorflow.contrib.losses.python.losses.loss_ops) is deprecated and will be removed after 2016-12-30.
Instructions for updating:
Use tf.losses.compute_weighted_loss instead.
WARNING:tensorflow:From \AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\contrib\losses\python\losses\loss_ops.py:151: add_loss (from tensorflow.contrib.losses.python.losses.loss_ops) is deprecated and will be removed after 2016-12-30.
Instructions for updating:
Use tf.losses.add_loss instead.

**Comment by pusj** [comment]
I should add that quite a few of the above warnings are easy to solve, however, I have not been able to replace Monitors with SessionRunHook and I have not obtained the desired effect with SKCombat when creating the regressor.

**Comment by surjithbs17** [comment]
@pusj 
Looks like instructions to solve the warnings are given right below the warnings. For example 

WARNING:tensorflow:From \AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\contrib\learn\python\learn\models.py:107: mean_squared_error_regressor (from tensorflow.contrib.learn.python.learn.ops.losses_ops) is deprecated and will be removed after 2016-12-01.

> Instructions for updating:
> Use tf.contrib.losses.mean_squared_error and explicit logits computation.

**Comment by pusj** [comment]
@surjithbs17 ,

Yes, you are correct but I haven't bothered about those easy ones ;-)

As you can see I have tried working with e.g. the Validation monitor (which should have been removed already)...I have tried with SessionRunHook but no success there yet (it is sometimes difficult to see the full picture when reading the TF release notes)

**Comment by surjithbs17** [comment]
@pusj ,
Totally agreed.

**Comment by pusj** [comment]
@surjithbs17 

I've searched for examples on how to code this (real examples are much easier to learn from compared to release notes). I haven't gotten many hits worth mentioning while searching, this is not ideal but at least this is somewhere to start building on: https://github.com/ppwwyyxx/tensorpack/issues/147

**Comment by besirkurtulmus** [comment]
I've implemented the codebase on algorithmia here: https://algorithmia.com/algorithms/deeplearning/TensorflowLSTM

It seems to be working, but I've also had various issues, like working with larger datasets, and getting NaN loss errors etc.

For now it works for simple datasets with reduced size.

**Comment by surjithbs17** [comment]
@besirkurtulmus , I wanted to point out this. Usually this NaN losses will come when your dataset is not normalized (Lot of issues can be solved when the data is normalized properly to 0.0-1.0).

I am not sure whether it is normalized in your dataset. Is it normalized?
